### PR TITLE
Additional Queries for TcCaseAuditService

### PR DESCRIPTION
**Pull request by Gerromie**
**Story**
The user may want to have a more refined view of the live apps audit trail using additional options such as caseType, creationTime, and to have the results sorted.

**Changes**
1. Added `@Input()` to LiveAppsCaseAuditComponent to take creationTime, caseType, and orderBy
1. Modified the query parameters used when building the audiEvents URL to include the new options

**Tests**
It would be practical to test the API construction and use before publishing.

### CODE REVIEW

File: projects/tibco-tcstk/tc-liveapps-lib/src/lib/services/tc-case-audit.service.ts [hunk 1]
```diff
@@ -11,17 +11,32 @@ export class TcCaseAuditService {
 
   constructor(private http: HttpClient) { }
 
-  public getCaseAudit(caseRef: string, sandboxId: number, startAt: string, top: number): Observable<AuditEventList> {
+  public getCaseAudit(caseRef: string, sandboxId: number, startAt: string, top: number,
+    creationTime: string, caseType: CaseTypeAudit, orderBy: OrderBy): Observable<AuditEventList> {
```
Comment (jezrsmith) [inline projects/tibco-tcstk/tc-liveapps-lib/src/lib/services/tc-case-audit.service.ts:15]: Suggest making the extra parameters optional so we don't break existing apps.   eg:) public getCaseAudit(caseRef: string, sandboxId: number, startAt: string, top: number,
    creationTime?: string, caseType?: CaseTypeAudit, orderBy?: OrderBy): Observable<AuditEventList> {

Comment (jezrsmith) [reply projects/tibco-tcstk/tc-liveapps-lib/src/lib/services/tc-case-audit.service.ts:15]: Also only add to URL if they are passed...

**Review by jezrsmith — CHANGES_REQUESTED** [review]
Can we make extra params optional (use ?) and then check they exist before adding to URL. that will ensure we dont break any other code using the lib

File: projects/tibco-tcstk/tc-liveapps-lib/src/lib/services/tc-case-audit.service.ts (hunk 1 above)
Comment (Gerromie) [reply projects/tibco-tcstk/tc-liveapps-lib/src/lib/services/tc-case-audit.service.ts]: Feedback regarding optional values accepted and pushed in.

Isn't the code below sufficient at detecting if they values exist before adding it to the url? See lines 19, 21, and 24 below.

**Review by jezrsmith — APPROVED** [review]

**Pull request merged**